### DISCUSSION TITLE
WebView2 Viewer plugin with full Markdown

### DISCUSSION THREAD

**Discussion by melloware** [OpenSalamander/salamander#16] [Ideas]
Currently the Markdown plugin uses the IE plugin and can't render images or full Markdown.  I would love to see a native Markdown plugin for *.md files using portable C or C++ code.

## CMark 
https://github.com/commonmark/cmark

**Advantages:**

- **Portable.** The library and program are written in standard C99 and have no external dependencies. They have been tested with MSVC, gcc, tcc, and clang.
- **Fast.** cmark can render a Markdown version of War and Peace in the blink of an eye (127 milliseconds on a ten year old laptop, vs. 100-400 milliseconds for an eye blink). In our [benchmarks](https://github.com/commonmark/cmark/blob/master/benchmarks.md), cmark is 10,000 times faster than the original Markdown.pl, and on par with the very fastest available Markdown processors.
- **Accurate.** The library passes all CommonMark conformance tests.
- **Standardized.** The library can be expected to parse CommonMark the same way as any other conforming parser. So, for example, you can use commonmark.js on the client to preview content that will be rendered on the server using cmark.
- **Robust.** The library has been extensively fuzz-tested using [american fuzzy lop](http://lcamtuf.coredump.cx/afl/). The test suite includes pathological cases that bring many other Markdown parsers to a crawl (for example, thousands-deep nested bracketed text or block quotes).
- **Flexible.** CommonMark input is parsed to an AST which can be manipulated programmatically prior to rendering.
- **Multiple renderers.** Output in HTML, groff man, LaTeX, CommonMark, and a custom XML format is supported. And it is easy to write new renderers to support other formats.
- **Free.** BSD2-licensed.

## Discount 
https://github.com/Orc/discount

**Advantages:**
- Extremely fast
- Written in pure C (no 3rd part dependencies).
- Actively maintained
- Has permissive 3 clause BSD license

**Comment by janrysavy** [comment]
Are you proposing to create a new plugin on top of Microsoft Edge WebView2?

Are you interested in doing it?

**Comment by melloware** [comment]
@janrysavy I am not very experience in C/C++ so i fear i would make a mess of it trying. I was putting this ticket here as I was hoping enhancement tickets would go here instead of in the forum.

I have been with you guys since 1999 and **I LOVE THIS TOOL**!!!

**Comment by janrysavy** [comment]
That's fine, it's a good idea, thank you. I've long considered ditching IE Viewer and switching to MS Edge WebView2. Better support for Markdown would be great.

**Comment by melloware** [comment]
I have another idea too if you don't mind me opening another enhancement plugin?

**Comment by KRtekTM** [comment]
https://github.com/KRtkovo-eu-AI/salamander/tree/plugins_darkmode/src/plugins/webview2renderviewer

<img width="1426" height="736" alt="image" src="https://github.com/user-attachments/assets/bcba4668-7688-4f16-904a-2769a232c3e0" />

# WebView2 Render Viewer .NET Supported File Types

| Category | Extensions |
| --- | --- |
| HTML and web archives | `.html`, `.htm`, `.xhtml`, `.mhtml`, `.mht` |
| Markdown (rendered to HTML) | `.md`, `.markdown`, `.mdown`, `.mkd`, `.mdx` |
| SVG vector graphics | `.svg`, `.svgz` |
| Modern image formats | `.webp`, `.avif`, `.apng` |
| Raster image formats | `.png`, `.jpg`, `.jpeg`, `.jfif`, `.gif`, `.bmp`, `.ico`, `.tif`, `.tiff` |
| Portable Document Format | `.pdf` |

All matches are case-insensitive. Files larger than 32 MB are rejected to match the plug-in limits.
